baseapp-profiles/baseapp_profiles/models.py

```diff
@@ -109,17 +110,37 @@ def get_all_users(self):
             models.Q(profiles_owner=self) | models.Q(profile_members__profile=self)
         ).distinct()
 
-    def generate_url_path(self, random=False):
-        if random:
-            return f"profile/{uuid.uuid4()}"
-        return f"profile/{self.pk}"
+    def generate_url_path(self, current_path_string=None):
+        if current_path_string:
+            last_char = current_path_string[-1]
+            if last_char.isdigit():
+                current_path_string = current_path_string[:-1] + str(int(last_char) + 1)
+            else:
+                current_path_string = current_path_string + "1"
+            return current_path_string
+
+        name = self.name or ""
+        # Remove whitespaces
+        name = name.translate(str.maketrans("", "", string.whitespace))
+
+        # If name is an email (which would only occur if the user's first and last names are empty during user registration),
+        # we'll remove the email domain and check if it's less than 8 characters. If it is, we'll add random digits to make it 8 characters.
+        # OBS: We're not checking for any other special chars since we've blocked it in the RegisterSerializer in baseapp-auth. If
+        # that changes, we should add a check here.
+        name = name.split("@")[0]
+        if len(name) < 8:
+            name = name + "".join(random.choices(string.digits, k=8 - len(name)))
+
+        return f"/{name}"
 
     def create_url_path(self):
         from baseapp_pages.models import URLPath
 
         url_path = self.generate_url_path()
-        if URLPath.objects.filter(path=url_path).exists():
-            url_path = self.generate_url_path(random=True)
+        is_duplicate = URLPath.objects.filter(path=url_path).exists()
+        while is_duplicate:
```

Comment:
the previous version was already recursive, I think this one can lead to multiple while loops running inside the other. not a big problem but I think a simple if is better here